packages/conferer/src/Conferer/Config/Internal.hs

```diff
@@ -166,15 +166,15 @@ findAndSplitList cond list = go [] list
           go (curElem:prevElems) nextElems
 
 -- | This function gets a value from 'Source's but ignores mappings and defaults
-getRawKeyInSources :: Key -> Config -> IO (Maybe (Key, Text))
+getRawKeyInSources :: Key -> Config -> IO (Maybe (Key, Int, Text))
 getRawKeyInSources k Config{..} =
-  go configSources
+  go $ zip [0..] configSources
```

Comment:
💭 Have you considered using something like a map from key (I think?) to Source instead of passing the index around?